components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/DataMapperMediator.java

```diff
@@ -392,23 +434,73 @@ private MappingResource getMappingResource(MessageContext synCtx, String configK
-     * @param key    registry key
-     * @return mapping configuration, inputSchema and outputSchema as inputStreams
+     * @param propertiesNamesList Required properties
+     * @param synCtx              Message context
+     * @return Map filed with property name and the value
      */
-    private static InputStream getRegistryResource(MessageContext synCtx, String key) {
-        InputStream inputStream = null;
-        Object entry = synCtx.getEntry(key);
-        if (entry instanceof OMTextImpl) {
-            if (log.isDebugEnabled()) {
-                log.debug("Value for the key is ");
+    private Map getPropertiesMap(List<String> propertiesNamesList, MessageContext synCtx) {
+        Map<String, Map<String, String>> propertiesMap = new HashMap<>();
+        String[] contextAndName;
+        String value = "";
+        org.apache.axis2.context.MessageContext axis2MsgCtx = ((Axis2MessageContext) synCtx).getAxis2MessageContext();
+        HashMap functionProperties = new HashMap();
+        Stack<TemplateContext> templeteContextStack = ((Stack) synCtx
+                .getProperty(SynapseConstants.SYNAPSE__FUNCTION__STACK));
+        if (templeteContextStack != null) {
+            TemplateContext templateContext = templeteContextStack.peek();
+            functionProperties.putAll(templateContext.getMappedValues());
+        }
+        for (String propertyName : propertiesNamesList) {
+            contextAndName = propertyName.split("\\['|'\\]");
+            switch (contextAndName[0].toUpperCase()) {
+            case DEFAULT_CONTEXT:
+            case SYNAPSE_CONTEXT:
+                value = (String) synCtx.getProperty(contextAndName[1]);
+                break;
+            case TRANSPORT_CONTEXT:
+                value = (String) ((Map) axis2MsgCtx.getProperty(TRANSPORT_HEADERS)).get(contextAndName[1]);
+                break;
+            case AXIS2_CONTEXT:
+                value = (String) axis2MsgCtx.getProperty(contextAndName[1]);
+                break;
+            case AXIS2_CLIENT_CONTEXT:
+                value = (String) axis2MsgCtx.getOptions().getProperty(contextAndName[1]);
+                break;
+            case OPERATIONS_CONTEXT:
+                value = (String) axis2MsgCtx.getOperationContext().getProperty(contextAndName[1]);
+                break;
+            case FUNCTION_CONTEXT:
+                value = (String) functionProperties.get(contextAndName[1]);
+                break;
+            default:
+                log.warn(contextAndName[0] + " scope is not found. Setting it to an empty value.");
+                value = "";
             }
-            OMTextImpl text = (OMTextImpl) entry;
-            String content = text.getText();
-            inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+            if (value == null) {
+                log.warn(propertyName + "not found. Setting it to an empty value.");
+                value = "";
```

Comment:
"" make this as a constant, you have used it several times in the code.. this is default value or something